src/styles/main.scss

```diff
@@ -1,3 +1,82 @@
+html {
+  font-family: Arial, Helvetica, sans-serif;
+  font-size: 30px;
+}
+
 body {
   margin: 0;
 }
+
+@function calendarWidth($gap, $size, $padding) {
+
+  @return ($size * 7) + ($gap * 6) + ($padding * 2);
+
+}
+
+$month-length: 31;
+
+$day-size: 100px;
+
+$gap-size: 1px;
+
+$padding-size: 10px;
+
+$day-name: 
+  "mon" 0,
+  "tue" 1,
+  "wed" 2,
+  "thu" 3,
+  "fri" 4,
+  "sat" 5,
+  "sun" 6;
+
+@mixin square($size) {
+  width: $size;
+  height: $size;
+}
+
+.calendar {
+  max-width: calendarWidth($gap-size, $day-size, $padding-size);
+  margin: 0 auto;
+  padding: $padding-size;
+  display: flex;
+  flex-direction: row;
+  flex-wrap: wrap;
+  gap: $gap-size;
+  
+  
+  @each $day, $value in $day-name {
+    &__start-day--#{$day} :first-child {
+      margin-left: (($day-size + $gap-size) * $value);
+    }
+  }
+
+  &__day {
+    background-color: #eee;
+    @include square($day-size);
+    box-sizing: border-box;
+    border: 1px solid #000;
+  }
+
+  @for $index from 1 through 31 {
+    &__day:nth-child(n + #{$index}):before {
+      content: "#{$index}";
+      display: flex;
+      height: 100%;
+      align-items: center;
+      justify-content: center;
+    }
+  }
+  @for $date from 28 through $month-length {
+    &__month-length--#{$date} :nth-child(n + #{$date + 1}) {
+      display: none;
+    }
+  }
+
+  &__day:hover {
+    cursor: pointer;
+    transform: translateY(-20px);
+    background-color: #ffc0cb;
+    transition-duration: 0.5s;
```

Comment:
add this transition-duration directly to `.calendar__day` to make your transition smoothly when you remove cursor from item